parteh/PRTParamsFATESMod.F90

```diff
@@ -1383,8 +1392,154 @@ subroutine PRTCheckParams(is_master)
+              !   Calculations for the generic case require allom_h2cd1 to be positive. If
+              ! not, issue an error.
+              ! ----------------------------------------------------------------------------
+              write(fates_log(),*) "---~---"
+              write(fates_log(),*) " Incorrect settings for crown depth allometry."
+              write(fates_log(),*) ' PFT index:   ',ipft
+              write(fates_log(),*) ' allom_dmode: ',prt_params%allom_dmode(ipft)
+              write(fates_log(),*) ' allom_h2cd1: ',prt_params%allom_h2cd1(ipft)
+              write(fates_log(),*) " Parameter ""allom_h2cd1"" must be positive when"
+              write(fates_log(),*) "    allom_dmode = 2."
+              write(fates_log(),*) "---~---"
+              write(fates_log(),*) ''
+              write(fates_log(),*) ''
+              nerror = nerror + 1
+           elseif ( abs(prt_params%allom_h2cd2 (ipft) - 1._r8 ) < nearzero ) then
+              !    Special case when the log-linear equation reduces to linear. This must
+              ! be checked separately to avoid singularities in the general case.
+              ! ----------------------------------------------------------------------------
+              if ( ( prt_params%allom_h2cd1 (ipft) < nearzero ) .or. &
+                   ( prt_params%allom_h2cd1 (ipft) > 1._r8    ) ) then
+                 write(fates_log(),*) "---~---"
+                 write(fates_log(),*) " Incorrect settings for crown depth allometry."
+                 write(fates_log(),*) ' PFT index:   ',ipft
+                 write(fates_log(),*) ' allom_dmode: ',prt_params%allom_dmode(ipft)
+                 write(fates_log(),*) ' allom_h2cd1: ',prt_params%allom_h2cd1(ipft)
+                 write(fates_log(),*) " Parameter ""allom_h2cd1"" must be positive and <= 1"
+                 write(fates_log(),*) "    when allom_h2cd2 = 1 (or allom_dmode = 1)."
+                 write(fates_log(),*) "---~---"
+                 write(fates_log(),*) ''
+                 write(fates_log(),*) ''
+                 nerror = nerror + 1
+              end if
+           elseif (is_hmode_fine) then
+              ! ----------------------------------------------------------------------------
+              !    General log-linear case. Depending on the parameter values, crown depth 
+              ! may exceed for very small or very large plants. The code has safeguards 
+              ! to prevent this behaviour, but we must at least issue some warning.
+              ! ----------------------------------------------------------------------------
+
+              !   Find the critical height in which crown depth becomes height
+              ! ----------------------------------------------------------------------------
+              height_crit = prt_params%allom_h2cd1 (ipft) ** &
+                           ( 1.0_r8 / (1.0_r8 - prt_params%allom_h2cd2 (ipft)) )
```

Comment:
Since we don't end the run till the bottom of this section, it's possible that we could have `is_hode_fine = true` and `allom_h2cd2 = 1.0` which we result in a divide by zero here, correct?

Reply:
> Since we don't end the run till the bottom of this section, it's possible that we could have `is_hode_fine = true` and `allom_h2cd2 = 1.0` which we result in a divide by zero here, correct?

I think this should never happen, because when `prt_params%allom_h2cd2 (ipft) = 1.0_r8`, then the code would enter in the `elseif` block right before this one instead (the one that checks `abs(prt_params%allom_h2cd2 (ipft) - 1._r8 ) < nearzero`).  

I added that previous block to avoid the singularity, and also because that case is numerically the same case as the linear crown depth allometry (`prt_params%allom_dmode(ipft)) =1`), so the code runs the same checks.

Reply:
Gotcha, I see now.  I overlooked that the previous `elseif` statement was in the same `if..end if` chain.  Thanks!